addons/account/models/account_move.py

```diff
@@ -2517,6 +2524,13 @@ def _get_last_sequence_domain(self, relaxed=False):
                 where_string += " AND date_trunc('year', date::timestamp without time zone) = date_trunc('year', %(date)s) "
                 param['date'] = self.date
                 param['anti_regex'] = re.sub(r"\?P<\w+>", "?:", self._sequence_monthly_regex.split('(?P<seq>')[0]) + '$'
+            elif sequence_number_reset == 'fyear_start':
+                company = self.company_id
+                fyear_start, fyear_end = date_utils.get_fiscal_year(self.date, day=company.fiscalyear_last_day, month=int(company.fiscalyear_last_month))
+                where_string += """ AND date BETWEEN %(fyear_start)s AND %(fyear_end)s"""
+                param['fyear_start'] = fyear_start
+                param['fyear_end'] = fyear_end
+                param['anti_regex'] = re.sub(r"\?P<\w+>", "?:", self._sequence_monthly_regex.split('(?P<seq>')[0]) + '$'
```

Comment:
These 2 regexes are conflicting too hard

Reply:
help me with this

Reply:
Add an assertion in `test_sequence_get_more_specific` to make sure that it works (look at the related commit if needed)

Reply:
You probably want to compare it to `_sequence_yearly_regex` though